RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -399,54 +430,64 @@ namespace mkfit {
       const auto ntracks = tracks.size();
 
       std::vector<float> ctheta(ntracks);
+      std::multimap<int, int> theHitMap;
+
       for (auto itrack = 0U; itrack < ntracks; itrack++) {
         auto &trk = tracks[itrack];
         ctheta[itrack] = 1.f / std::tan(trk.theta());
+        for (int i = 0; i < trk.nTotalHits(); ++i) {
+          if (trk.getHitIdx(i) < 0)
+            continue;
+          int a = trk.getHitLyr(i);
+          int b = trk.getHitIdx(i);
+          theHitMap.insert(std::make_pair(b * 1000 + a, i > 0 ? itrack : -itrack));  //negative for first hit in trk
+        }
       }
 
       for (auto itrack = 0U; itrack < ntracks; itrack++) {
         auto &trk = tracks[itrack];
         auto phi1 = trk.momPhi();
         auto ctheta1 = ctheta[itrack];
 
-        for (auto jtrack = itrack + 1; jtrack < ntracks; jtrack++) {
-          auto &track2 = tracks[jtrack];
-          auto sharedCount = 0;
-          auto sharedFirst = 0;
+        std::map<int, int> theSharingMap;
+        for (int i = 0; i < trk.nTotalHits(); ++i) {
+          if (trk.getHitIdx(i) < 0)
+            continue;
+          int a = trk.getHitLyr(i);
+          int b = trk.getHitIdx(i);
+          auto range = theHitMap.equal_range(b * 1000 + a);
+          for (auto it = range.first; it != range.second; ++it) {
+            if (std::abs(it->second) >= (int)itrack)
+              continue;  // don't check your own hits (==) nor sym. checks (>)
+            if (i == 0 && it->second < 0)
+              continue;  // shared first - first is not counted
+            theSharingMap[std::abs(it->second)]++;
+          }
+        }
 
-          auto dctheta = std::abs(ctheta[jtrack] - ctheta1);
+        for (const auto &elem : theSharingMap) {
+          auto &track2 = tracks[elem.first];
+
+          //these dctheta dphi are wasting time here -> need to remove these checks and revisit the fraction WP
```

Comment:
considering the past review comments, some rephrasing may be useful; or just drop the comment.
e.g. `broad dctheta-dphi compatibility checks; keep mostly to preserve consistency with old results`